Stratis.Bitcoin/RPC/Controllers/FullNodeController.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using NBitcoin;
+using Stratis.Bitcoin.RPC.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Stratis.Bitcoin.RPC.Controllers
+{
+	//TODO: Need to be extensible, should be FullNodeController
+	public partial class ConsensusController : Controller
+	{
+		[ActionName("stop")]
+		public Task Stop()
+		{
+			_FullNode.Dispose();
+			return Task.CompletedTask;
+		}
+
+		[ActionName("getrawtransaction")]
+		public async Task<TransactionModel> GetRawTransaction(string txid, int verbose = 0)
+		{
+			uint256 trxid;
+			if (!NBitcoin.uint256.TryParse(txid, out trxid))
+				throw new ArgumentException(nameof(txid));
+
+			ChainedBlock block = null;
+			Transaction trx = (await _FullNode.MempoolManager.InfoAsync(trxid))?.Trx;
+			if (trx == null)
+			{
+				var blockid = await _FullNode.BlockStoreManager.BlockRepository.GetTrxBlockIdAsync(trxid);
```

Comment:
Why not use BlockRepository.GetTrxAsync() and avoid the second call to the repo? (two thread grabs)